src/common/TrackedOp.h

```diff
@@ -190,19 +190,35 @@ class OpTracker {
   {
     typename T::Ref retval(new T(params, this));
     retval->tracking_start();
-
     if (is_tracking()) {
-      retval->mark_event("header_read", params->get_recv_stamp());
-      retval->mark_event("throttled", params->get_throttle_stamp());
-      retval->mark_event("all_read", params->get_recv_complete_stamp());
-      retval->mark_event("dispatched", params->get_dispatch_stamp());
+      struct MarkEventStore {
+	std::string str;
+	utime_t stamp;
+
+	MarkEventStore(std::string s, utime_t t) : str(s), stamp(t) {}
```

Comment:
please pass by `string_view`, a temporary instance of `std::string` is expensive.